server/monitor-conditions/operators.js

```diff
@@ -0,0 +1,201 @@
+class ConditionOperator {
+    id = undefined;
+    caption = undefined;
+
+    /**
+     * @type {mixed} variable
+     * @type {mixed} value
+     */
+    test(variable, value) {
+        throw new Error("You need to override test()");
+    }
+}
+
+const OP_EQUALS = "equals";
+
+const OP_NOT_EQUALS = "not_equals";
+
+const OP_CONTAINS = "contains";
+
+const OP_NOT_CONTAINS = "not_contains";
+
+const OP_LT = "lt";
+
+const OP_GT = "gt";
+
+const OP_LTE = "lte";
+
+const OP_GTE = "gte";
+
+/**
+ * Asserts a variable is equal to a value.
+ */
+class EqualsOperator extends ConditionOperator {
+    id = OP_EQUALS;
+    caption = "equals";
+
+    /**
+     * @inheritdoc
+     */
+    test(variable, value) {
+        // eslint-disable-next-line eqeqeq
+        return variable == value;
```

Comment:
I initially was going to add an `mx_priority` variable to check MX record priorities. Node's DNS module parses MX records and returns priority as a number type.

In this new conditions UI, comparison values are stored as strings. i.e. My condition would be stored in the monitor as `mx_priority equals "10"`.

I have a set of numeric operators (less than, greater than, etc) that cast the value to Number internally, but the EqualsOperator is currently used for both strings and numbers. It needs loose equality to handle 10 == "10".

An alternate solution could be to rename this operator to StringEquals (make it strict) and add a new NumberEquals operator (also strict) that casts value internally.

Reply:
Adding a comment that this is because of sharing between numbers and strings to handle `10 == "10"` is likely fine.

Reply:
I'm flip-flopping on this. I think it may be advantageous to start out with strict operators rather than maybe trying to implement them later on (and have to worry about backwards compatibility.)

So, require a monitor type to explicitly use either `StringEquals` or `NumberEquals` when declaring its supported condition variables.